packages/remix-node/stream.ts

```diff
@@ -0,0 +1,196 @@
+import type { Writable } from "stream";
+import { Stream } from "stream";
+
+const { readableHighWaterMark } = new Stream.Readable();
+
+export async function pipeReadableStreamToWritable(
+  stream: ReadableStream,
+  writable: Writable
+) {
+  let reader = stream.getReader();
+
+  async function read() {
+    let { done, value } = await reader.read();
+
+    if (done) {
+      writable.end();
+      return;
+    }
+
+    writable.write(value);
+
+    await read();
+  }
+
+  await read();
+}
+
+export async function readableStreamToBase64String(stream: ReadableStream) {
+  let reader = stream.getReader();
+  let chunks: Uint8Array[] = [];
+
+  async function read() {
+    let { done, value } = await reader.read();
+
+    if (done) {
+      return;
+    } else if (value) {
+      chunks.push(value);
+    }
+
+    await read();
+  }
+
+  await read();
+
+  return Buffer.concat(chunks).toString("base64");
+}
+
+export async function readableStreamToString(stream: ReadableStream<any>) {
```

Comment:
Add optional `encoding` arg for the `buffer.toString(encoding)` and just have one function instead of 2 (get rid of `readableStreamToBase64String`)

Reply:
maybe this belongs in server-runtime? but it can't use Buffer if it is.

Reply:
I'm going to say it should stay. Using the native Buffer is going to be significantly faster than doing this ourselves in userland.